### DISCUSSION TITLE
livewire.js Uncaught SyntaxError: Invalid or unexpected token

### DISCUSSION THREAD

**Discussion by NoxzMSTR** [livewire/livewire#6597] [1. Help] [answered]
Hi can you help me i installed the livewire 3 but when i add component the js file says error i know why because livewire generating it from backend it put white spaces which i dont find in my software. so how to work with it.
I copy the livewire js to public and put the asset_url on config and it worked fine but not auto generated

![image](https://github.com/livewire/livewire/assets/42188571/c7d01a1f-b197-43c1-8d7f-f703db07b5cb)

**Comment by NoxzMSTR** [comment]
Also when i added the livewire.js to public and use the asset_url in livewire config then i want to use the navigate functionality so when i navigate it show on top the bar but dont change the url and shows this error
![image](https://github.com/livewire/livewire/assets/42188571/2023ff53-99e1-4f4f-a935-d7f0e0ca0941)

**Reply by ChaerilM** [reply]
is Alpine js loaded?

**Comment by NoxzMSTR** [comment]
Sorry i got that fixed because of white spaces just next left is
alpine.destorytree function is not found issue

**Reply by abishekrsrikaanth** [reply]
I am getting the same error. What was the fix?

**Reply by NoxzMSTR** [reply, marked as answer]
The issue is spaces when doing request so i create  a function which gives json output then i check every route files for extra spaces then found some file with 

space
space
<?php start tag

so i removed it and fixed the problem but i got problem again with vite
Problem was that livewire was loaded first and vite is after so in the bootstrap.js there is apline js import so it is overwriting the livewire alpine js so the function not found

**Comment by jonerickson** [comment]
If anyone struggling with this error, I found the cause of mine - simply just an extra space at the top of one of my PHP files. The opening PHP tag started on line 2 instead of 1.

**Reply by mariocosttaa** [reply]
Obrigado, trabalhou pra mim!

**Comment by IbrahimEzat** [comment]
I struggled with this error for a while, and finally by isolating the code chunk by chunk, I found that any comment inside JS code will cause such an error.

```
@script
    <script>
          /**
           * this below code for bla bla 
          */
         console.log('test')
    </script>
@endscript
```
any comment have to be deleted to be 
```
@script
    <script>
         console.log('test')
    </script>
@endscript
```

**Reply by caliwa** [reply]
i dont understand you, i do not have any @script in blade files and my project does not work in apache xampp :(

**Reply by IbrahimEzat** [reply]
I just share my experience with the same error message, it may help someone in the same situation.
@script should be there as livewire 3 documentation , if you will use JavaScript inside the blade file 
 [documentation link](https://livewire.laravel.com/docs/javascript#using-javascript-in-livewire-components)

**Reply by magarrent** [reply]
Wow!!! That was it. I don't know if you know that @joshhanley, but is still happening on Livewire 3.3

Any comment inside a "@script" crashes the "update" livewire action:

```
@script
    <script>
        // Hello, this will crash
        console.log('Component script executed');
```

**Comment by LEstradioto** [comment]
Had same problem. I think I had two `npm run dev` running, one outside of docker somehow. Perhaps help others too!
